github-page/_guides/migrate-to-8.md

```diff
@@ -0,0 +1,81 @@
+---
+title: "Migrate to 8.0.0"
+classes: wide
+---
+
+These are the changes needed to migrate to version 8 of Typegoose.
+
+## Typings for document creation
+Document creation now has types, so some old code may fail. Note that now, you MUST specify which parameters are optional and which are mandatory in your classes with `?` and `!`. Note that no modifier means that the property will be assumed required.
+
+```ts
+// This is bad - both name and favoriteNumber are supposed to be mandatory, but appear mandatory to TypeScript
+class MyOldClass {
+  @prop();
+  name: string;
+
+  @prop();
+  favoriteNumber: number;
+}
+
+const MyOldClassModel = getModelForClass(MyOldClass);
+
+new MyOldClassModel({ }); // TypeScript complains
+new MyOldClassModel(); // Works, but it should not
+new MyOldClassModel({
+  name: 'Bob',
+  favoriteNumber: 23
+}); // Works
+```
+
+Usually, fixing is easy:
+
+```ts
+class MyNewClass {
+  @prop()
+  name?: string;
+
+  @prop()
+  favoriteNumber?: number;
+};
+
+const MyOldClassModel = getModelForClass(MyOldClass);
+
+new MyOldClassModel(); // Works, but should not
+MyOldClass.create(); // IMPORTANT: TYPESCRIPT ERROR
```

Comment:
```suggestion
await MyOldClass.create(); // IMPORTANT: TYPESCRIPT ERROR 
```

otherwise typescript might warn that it returns an un-awaited promise and to indicate to the user that this functions is awaitable

Reply:
We don't know if this is an asynchronous context. I would leave it as-is, most people know how to use promises with async/await.